### PR TITLE
[#119207915] saved credit card in checkout

### PR DESCRIPTION
**Pull request by CyberCookie**

### CODE REVIEW

File: src/app/checkout/accordion.controller.js [hunk 1]
```diff
@@ -102,10 +105,18 @@ angular.module('checkoutModule')
...
+                    tokens = res;
+                }
+            });
 
             // Load payment methods
             checkoutService.loadPaymentMethods().then(function(methods){
                 $scope.paymentMethods = methods;
+
+                //merge payment methods and tokens;
+                $scope.paymentMethods = $scope.paymentMethods.concat(tokens);
```
Comment (asavchenko) [inline src/app/checkout/accordion.controller.js]: tokens is a global to this function variable where does it come from?

File: src/app/checkout/service.js [hunk 2]
```diff
@@ -74,6 +74,18 @@ angular.module('checkoutModule')
                 return defer.promise;
             };
 
+            loadTokens = function () {
```
Comment (asavchenko) [inline src/app/checkout/service.js]: please add comment and describe what tokens are and what is a difference between token and payment method (word token can mean a lot of things)

File: src/app/checkout/accordion.controller.js [hunk 3]
```diff
@@ -38,6 +38,7 @@ angular.module('checkoutModule')
         $scope.cart = cartService;
         $scope.checkout = {};
         $scope.isGuestCheckout = false;
+        $scope.tokens = [];
```
Comment (asavchenko) [inline src/app/checkout/accordion.controller.js]: is this still needed?

File: src/app/checkout/accordion.controller.js [hunk 4]
```diff
@@ -102,10 +104,18 @@ angular.module('checkoutModule')
...
+                    savedCC = res;
+                }
+            });
 
             // Load payment methods
             checkoutService.loadPaymentMethods().then(function(methods){
                 $scope.paymentMethods = methods;
+
+                //merge payment methods and saved Credit Cards;
+                $scope.paymentMethods = $scope.paymentMethods.concat(savedCC);
```
Comment (dglushenkov) [inline src/app/checkout/accordion.controller.js]: I think we can't be sure that payments will be loaded after saved cc tokens. So what if payments are loaded and tokens are not yet? My suggestion is to concatenate `methods` and `savedCC` with `$scope.paymentMethods`

**Comment by asavchenko** [comment]
@dglushenkov looks good

**Comment by dglushenkov** [comment]
@asavchenko ok, merging

**Pull request merged**